cmd/metricsbridge/metricsbridge.go

```diff
@@ -0,0 +1,234 @@
+	default:
+		c.rootCAs = x509.NewCertPool()
+		c.rootCAs.AppendCertsFromPEM(b)
+	}
+
+	b, err = ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	switch {
+	case os.IsNotExist(err):
+	case err != nil:
+		return err
+	default:
+		c.Token = string(b)
+	}
+
+	return nil
+}
+
+func (c *config) validate() (errs []error) {
+	if c.Interval < time.Second {
+		errs = append(errs, fmt.Errorf("intervalNanoseconds %q too small", c.Interval))
+	}
+	if _, err := url.Parse(c.PrometheusFederateEndpoint); err != nil {
+		errs = append(errs, fmt.Errorf("prometheusFederateEndpoint: %s", err))
+	}
+	if _, err := net.ResolveUDPAddr("udp", c.StatsdEndpoint); err != nil {
+		errs = append(errs, fmt.Errorf("statsdEndpoint: %s", err))
+	}
+	if len(c.Series) == 0 {
+		errs = append(errs, fmt.Errorf("must configure at least one series"))
+	}
+
+	return
+}
+
+func (c *config) init() error {
+	var err error
+
+	c.statsd, err = statsd.NewClient(c.StatsdEndpoint)
+	if err != nil {
+		return err
+	}
+
+	c.http = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:            c.rootCAs,
+				InsecureSkipVerify: c.InsecureSkipVerify,
+			},
+		},
+	}
+
+	return nil
+}
+
+func run(log *logrus.Entry, configpath string) error {
+	c := &config{log: log}
+
+	if err := c.load(configpath); err != nil {
+		return err
+	}
+
+	if errs := c.validate(); len(errs) > 0 {
+		var sb strings.Builder
+		for _, err := range errs {
+			sb.WriteString(err.Error())
+			sb.WriteByte('\n')
+		}
+		return errors.New(sb.String())
+	}
+
+	if err := c.init(); err != nil {
+		return err
+	}
+
+	return c.run()
+}
+
+func (c *config) run() error {
+	prometheusURL, err := url.Parse(c.PrometheusFederateEndpoint)
+	if err != nil {
+		return err
+	}
+
+	v := url.Values{}
+	for _, s := range c.Series {
+		v.Add("match[]", s)
+	}
+	prometheusURL.RawQuery = v.Encode()
+
+	req, err := http.NewRequest(http.MethodGet, prometheusURL.String(), nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Authorization", "Bearer "+c.Token)
+	req.Header.Add("Accept", string(expfmt.FmtText))
+
+	t := time.NewTicker(c.Interval)
+	defer t.Stop()
+
+	for {
+		if err := c.runOnce(req); err != nil {
+			c.log.Warn(err)
+		}
+		<-t.C
+	}
+}
+
+func (c *config) runOnce(req *http.Request) error {
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("prometheus returned status code %d", resp.StatusCode)
+	}
+
+	d := expfmt.NewDecoder(resp.Body, expfmt.FmtText)
+
+	for {
+		var family io_prometheus_client.MetricFamily
+
+		err = d.Decode(&family)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		for _, m := range family.Metric {
+			gauge := &statsd.Gauge{
+				Namespace: c.Namespace,
+				Metric:    *family.Name,
+				Dims: map[string]string{
+					"Region":       c.Region,
+					"UnderlayName": "",
+				},
+				Value: *m.Untyped.Value,
```

Comment:
```
GOOD:
........j-.{"Namespace":"Prometheus","Metric":"kube_node_info","Dims":{"Region":"eastus","UnderlayName":"","cluster":"https://mjudeikis-metrics.eastus.cloudapp.azure.com","container_runtime_version":"docker://1.13.1","endpoint":"https-main","instance":"10.131.0.14:8443","job":"kube
-state-metrics","kernel_version":"3.10.0-957.el7.x86_64","kubelet_version":"v1.11.0+d4cacc0","kubeproxy_version":"v1.11.0+d4cacc0","namespace":"openshift-monitoring","node":"master-000000","os_image":"Red Hat Enterprise Linux Server 7.6 (Maipo)","pod":"kube-state-metrics-64cc8b86b5
-xt948","prometheus":"openshift-monitoring/k8s","prometheus_replica":"prometheus-k8s-0","provider_id":"azure:///subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/mjudeikis-metrics/providers/Microsoft.Compute/virtualMachineScaleSets/ss-master/virtualMachines/0","serv
ice":"kube-state-metrics"}}:1|g

CURRENT:
........[".{"__name__":"kube_node_status_condition","cluster":"https://mjudeikis-metrics.eastus.cloudapp.azure.com","condition":"Ready","endpoint":"https-main","instance":"10.131.0.14:8443","job":"kube-state-metrics","namespace":"openshift-monitoring","node":"master-000002","pod":"
kube-state-metrics-64cc8b86b5-xt948","prometheus":"openshift-monitoring/k8s","prometheus_replica":"prometheus-k8s-1","service":"kube-state-metrics","status":"unknown"}:0.000000|g                                                                                                        
```

After speaking with @sozercan and some TCPDUMP'int - converted pod is converting all `float64` to `int64`. We could try not to convert and see how MDM handles those. From a metrics point of I would like to keep a float, as it gives more freedom in the calculation space.

Reply:
Ignore. Seen the conversion now.